rust/patchable/src/main.rs

```diff
@@ -414,13 +437,71 @@ fn main() -> Result<()> {
             // --base can be a reference, but patchable.toml should always have a resolved commit id,
             // so that it cannot be changed under our feet (without us knowing so, anyway...).
             tracing::info!(?base, "resolving base commit-ish");
-            let base_commit = repo::resolve_and_fetch_commitish(&product_repo, &base, &upstream)
-                .context(FetchBaseCommitSnafu)?;
+            let base_commit = repo::resolve_and_fetch_commitish(&product_repo, &base, &upstream).context(FetchBaseCommitSnafu)?;
+            let mut upstream_mirror = None;
+
+            if mirrored {
+                // Parse e.g. "https://github.com/apache/druid.git" into "druid"
+                let repo_name = upstream.split('/').last().map(|repo| repo.trim_end_matches(".git")).context(ParseUpstreamUrlSnafu { url: &upstream })?;
+
+                ensure!(!repo_name.is_empty(), ParseUpstreamUrlSnafu { url: &upstream });
+
+                let mirror_url = format!("https://github.com/stackabletech/{}.git", repo_name);
+                tracing::info!(%mirror_url, "using mirror repository");
+
+                // Add mirror remote
+                let mut mirror_remote = product_repo
+                    .remote_anonymous(&mirror_url)
+                    .context(AddMirrorRemoteSnafu { url: mirror_url.clone() })?;
+
+                // Push the base commit to the mirror
+                tracing::info!(commit = %base_commit, base = base, url = mirror_url, "pushing commit to mirror");
+                let mut callbacks = git2::RemoteCallbacks::new();
+                callbacks.credentials(|_url, username_from_url, _allowed_types| {
+                    git2::Cred::credential_helper(
+                        &git2::Config::open_default().unwrap(), // Use default git config
+                        _url,
+                        username_from_url,
+                    )
+                });
```

Comment:
Do we actually need this? Feels like a lot of faff just to get the default behaviour...

Reply:
Fair :/ Maybe worth a comment?